CMakeLists.txt

```diff
@@ -68,7 +58,113 @@ set_target_properties(WinPixEventRuntime PROPERTIES
+    )
+
+    FetchContent_Declare(
+        tinyexr
+        GIT_REPOSITORY https://github.com/syoyo/tinyexr.git
+        GIT_TAG        v1.0.7
+        SOURCE_DIR     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/tinyexr/
+    )
+    set(TINYEXR_BUILD_SAMPLE OFF CACHE BOOL "")
+    set(TINYEXR_USE_MINIZ OFF CACHE BOOL "")
+
+    FetchContent_Declare(
+        ktx
+        GIT_REPOSITORY https://github.com/KhronosGroup/KTX-Software.git
+        GIT_TAG        v4.2.1
+        SOURCE_DIR     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/KTX-Software/
+    )   
+    set(KTX_FEATURE_TESTS OFF CACHE BOOL "")
+    set(KTX_FEATURE_VULKAN OFF CACHE BOOL "")
+    set(KTX_FEATURE_GL_UPLOAD OFF CACHE BOOL "")
+    set(KTX_FEATURE_VK_UPLOAD OFF CACHE BOOL "")
+    set(KTX_FEATURE_TOOLS OFF CACHE BOOL "")
+    set(KTX_FEATURE_DOC OFF CACHE BOOL "")
+    set(KTX_FEATURE_STATIC_LIBRARY ON CACHE BOOL "")
+
+    FetchContent_Declare(
+        vulkan-headers
+        GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers.git
+        GIT_TAG        v1.3.263
+        SOURCE_DIR     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/Vulkan-Headers/
+    )
+
+    FetchContent_MakeAvailable(stb cgltf glm tinyobjloader tinyexr ktx vulkan-headers)
+
+    target_include_directories(gfx PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/third_party/stb)
+    target_include_directories(gfx PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/third_party/cgltf)
+    target_include_directories(gfx PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/third_party/Vulkan-Headers/include)
+    target_include_directories(gfx PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/third_party/tinyexr)
+    target_link_libraries(gfx PUBLIC glm::glm)
+    target_link_libraries(gfx PRIVATE tinyobjloader ktx)
+
+    target_compile_definitions(gfx PUBLIC GFX_ENABLE_SCENE)
+
+    # Organize third party projects
+    set_target_properties(tinyobjloader PROPERTIES FOLDER "third_party")
+    set_target_properties(obj_basisu_cbind PROPERTIES FOLDER "third_party/ktx_deps")
+    set_target_properties(objUtil PROPERTIES FOLDER "third_party/ktx_deps")
+    set_target_properties(uninstall PROPERTIES FOLDER "third_party")
+    set_target_properties(tinyexr PROPERTIES FOLDER "third_party")
+    set_target_properties(ktx PROPERTIES FOLDER "third_party")
+    set_target_properties(ktx_read PROPERTIES FOLDER "third_party/ktx_deps")
+    set_target_properties(astcenc-avx2-static PROPERTIES FOLDER "third_party/ktx_deps")
+    set_target_properties(ktx_version PROPERTIES FOLDER "third_party/ktx_deps")
+    set_target_properties(fmt PROPERTIES FOLDER "third_party/ktx_deps")
+
+    if(TARGET mkvk)
+        set_target_properties(mkvk PROPERTIES FOLDER "third_party/ktx_deps")
+        set_target_properties(makedfd2vk PROPERTIES FOLDER "third_party/ktx_deps")
+        set_target_properties(makevk2dfd PROPERTIES FOLDER "third_party/ktx_deps")
+        set_target_properties(mkvkformatfiles PROPERTIES FOLDER "third_party/ktx_deps")
+    endif()
```

Comment:
visual studio project organisation was moved from capsaicin to here as then it applies to all of them and then to any other project using gfx